tests/integration-tests/clusters_factory.py

```diff
@@ -30,11 +30,35 @@ def __init__(self, name, config_file, ssh_key):
         self.__cfn_outputs = None
         self.__cfn_resources = None
 
-    def update(self):
-        """Rewrite the configuration file starting from the self.config object."""
+    def update(self, reset_desired=None, extra_params=None):
+        """
+        Update a cluster with an already updated config.
+        :param reset_desired: reset the current ASG desired capacity to initial config values
+        :param extra_params: extra parameters to pass to stack update
+        """
+        # Rewrite the configuration file starting from the already updated self.config object
         with open(self.config_file, "w") as configfile:
             self.config.write(configfile)
 
+        # update the cluster
+        logging.info("Updating cluster {0} with config {1}".format(self.name, self.config_file))
+        command = ["pcluster", "update", "--config", self.config_file]
+        if reset_desired:
+            command.append("--reset-desired")
+        if extra_params:
+            command.extend(["--extra-parameters", extra_params])
```

Comment:
if extra_params is a list this might not work as expected

Reply:
extra_params is a json, so a string